OpenRA.Mods.Common/Widgets/Logic/Ingame/CommandBarLogic.cs

```diff
@@ -0,0 +1,274 @@
+			if (forceMoveButton != null)
+			{
+				BindButtonIcon(forceMoveButton);
+
+				forceMoveButton.IsDisabled = () => { UpdateStateIfNecessary(); return forceMoveDisabled; };
+				forceMoveButton.IsHighlighted = () => !forceMoveButton.IsDisabled() && IsForceModifiersActive(Modifiers.Alt);
+				forceMoveButton.OnClick = () =>
+				{
+					if (forceMoveButton.IsHighlighted())
+						world.CancelInputMode();
+					else
+						world.OrderGenerator = new ForceModifiersOrderGenerator(Modifiers.Alt, true);
+				};
+			}
+
+			var forceAttackButton = widget.GetOrNull<ButtonWidget>("FORCE_ATTACK");
+			if (forceAttackButton != null)
+			{
+				BindButtonIcon(forceAttackButton);
+
+				forceAttackButton.IsDisabled = () => { UpdateStateIfNecessary(); return forceAttackDisabled; };
+				forceAttackButton.IsHighlighted = () => !forceAttackButton.IsDisabled() && IsForceModifiersActive(Modifiers.Ctrl);
+				forceAttackButton.OnClick = () =>
+				{
+					if (forceAttackButton.IsHighlighted())
+						world.CancelInputMode();
+					else
+						world.OrderGenerator = new ForceModifiersOrderGenerator(Modifiers.Ctrl, true);
+				};
+			}
+
+			var guardButton = widget.GetOrNull<ButtonWidget>("GUARD");
+			if (guardButton != null)
+			{
+				BindButtonIcon(guardButton);
+
+				guardButton.GetKey = _ => ks.GuardKey;
+				guardButton.IsDisabled = () => { UpdateStateIfNecessary(); return guardDisabled; };
+				guardButton.IsHighlighted = () => world.OrderGenerator is GenericSelectTarget
+					&& ((GenericSelectTarget)world.OrderGenerator).OrderName == "Guard";
+
+				guardButton.OnClick = () =>
+				{
+					if (guardButton.IsHighlighted())
+						world.CancelInputMode();
+					else
+						world.OrderGenerator = new GuardOrderGenerator(selectedActors,
+							"Guard", "guard", Game.Settings.Game.MouseButtonPreference.Action);
+				};
+			}
+
+			var scatterButton = widget.GetOrNull<ButtonWidget>("SCATTER");
+			if (scatterButton != null)
+			{
+				BindButtonIcon(scatterButton);
+
+				scatterButton.GetKey = _ => ks.ScatterKey;
+				scatterButton.IsDisabled = () => { UpdateStateIfNecessary(); return scatterDisabled; };
+				scatterButton.IsHighlighted = () => scatterHighlighted > 0;
+				scatterButton.OnClick = () =>
+				{
+					scatterHighlighted = 2;
+					PerformKeyboardOrderOnSelection(a => new Order("Scatter", a, false));
+				};
+			}
+
+			var deployButton = widget.GetOrNull<ButtonWidget>("DEPLOY");
+			if (deployButton != null)
+			{
+				BindButtonIcon(deployButton);
+
+				deployButton.GetKey = _ => ks.DeployKey;
+				deployButton.IsDisabled = () => { UpdateStateIfNecessary(); return !selectedDeploys.Any(Exts.IsTraitEnabled); };
+				deployButton.IsHighlighted = () => deployHighlighted > 0;
+				deployButton.OnKeyPress = ki =>
+				{
+					if (ki.Event == KeyInputEvent.Down && !ki.IsRepeat)
+						deployButton.OnClick();
+				};
+
+				deployButton.OnClick = () =>
+				{
+					deployHighlighted = 2;
```

Comment:
I noticed that there is additonal blick due to this on mouse click. The button is highlighted already on button click. This should be moved to OnKeyPress (also for Scatter, Stop).